Sources/Build/ManifestBuilder.swift

```diff
@@ -268,12 +285,147 @@ extension LLBuildManifestBuilder {
+
+        // Outputs.
+        let objectNodes = description.objects.map(Node.file)
+        let moduleNode = Node.file(description.moduleOutputPath)
+        let cmdOutputs = objectNodes + [moduleNode]
+
+        // Commands.
+        addExplicitBuildSwiftCmds(description, inputs: inputs,
+                                  objectNodes: objectNodes,
+                                  moduleNode: moduleNode,
+                                  targetDepGraphMap: &targetDepGraphMap)
+
+        addTargetCmd(description, cmdOutputs: cmdOutputs)
+        addModuleWrapCmd(description)
+    }
+
+    private func addExplicitBuildSwiftCmds(
+        _ targetDescription: SwiftTargetBuildDescription,
+        inputs: [Node], objectNodes: [Node], moduleNode: Node,
+        targetDepGraphMap: inout [ResolvedTarget: InterModuleDependencyGraph]
+    ) {
+        do {
+            // Pass the driver its external dependencies (target dependencies)
+            let targetDependencyMap = collectTargetDependencyInfos(for: targetDescription.target,
+                                                                   targetDepGraphMap: targetDepGraphMap)
+
+            // Compute the set of frontend
+            // jobs needed to build this Swift target.
+            var commandLine = targetDescription.emitCommandLine();
+            commandLine.append("-driver-use-frontend-path")
+            commandLine.append(buildParameters.toolchain.swiftCompiler.pathString)
+            commandLine.append("-experimental-explicit-module-build")
+            // FIXME: At some point SwiftPM should provide its own executor for
+            // running jobs/launching processes during planning
+            let executor = try SwiftDriverExecutor(diagnosticsEngine: plan.diagnostics,
+                                                   processSet: ProcessSet(),
+                                                   fileSystem: targetDescription.fs,
+                                                   env: ProcessEnv.vars)
+            var driver = try Driver(args: commandLine, fileSystem: targetDescription.fs,
+                                    executor: executor,
+                                    externalModuleDependencies: targetDependencyMap)
+
+            let jobs = try driver.planBuild()
+
+            // Save the dependency graph of this target to be used by its dependents
+            guard let dependencyGraph = driver.interModuleDependencyGraph else {
+                fatalError("Expected module dependency graph for target: \(targetDescription)")
+            }
+            targetDepGraphMap[targetDescription.target] = dependencyGraph
+            addSwiftDriverJobs(for: targetDescription, jobs: jobs, inputs: inputs,
+                               isMainModule: { driver.isExplicitMainModuleJob(job: $0)})
          } catch {
              fatalError("\(error)")
          }
     }
 
+    private func collectTargetDependencyInfos(for target: ResolvedTarget,
```

Comment:
Could you comment this function? It'd be worth pointing out, for example, that this is the transitive list of all dependencies. Giving the result type a `typealias` with a meaningful name would help a lot. I see, for example, that the `AbsolutePath` is the `.swiftmodule` file? Might we need to consider that ` .swiftinterface` would end up here?